src/core/symbology-ng/qgsgraduatedsymbolrendererv2.cpp

```diff
@@ -1386,9 +1399,14 @@ void QgsGraduatedSymbolRendererV2::updateSymbols( QgsSymbolV2 *sym )
   int i = 0;
   foreach ( QgsRendererRangeV2 range, mRanges )
   {
-    QgsSymbolV2 *symbol = sym->clone();
-    symbol->setColor( range.symbol()->color() );
-    updateRangeSymbol( i, symbol );
+    QScopedPointer<QgsSymbolV2> symbol( sym->clone() );
```

Comment:
I agree in principal, but this method only has a single exit point and no risk of leaks, so don't think the extra complexity is warranted.